src/pyxdf/pyxdf.py

```diff
@@ -29,8 +29,8 @@
 class StreamData:
     """Temporary per-stream data."""
 
-    def __init__(self, xml):
-        """Init a new StreamData object from a stream header."""
+    def __init__(self, xml, stream_id):
+        """Init a new StreamData object from a stream header and ID."""
```

Comment:
Is it really necessary to add a new parameter here? You only need the stream ID for logging, but you could extract it from the `xml` argument I think.

Reply:
OK, thanks! Since that seems to be a private function, passing `stream_id` should be no problem at all, and it allows us to keep the `StreamData` class clean.